tests/test_internals/test_si.py

```diff
@@ -585,66 +585,56 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
-class TestSpikeInterfaceRecorders(unittest.TestCase):
+class TestAddElectrodes(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
+        """Use common recording objects and values."""
+        cls.num_channels = 4
+        cls.base_recording = generate_recording(num_channels=cls.num_channels, durations=[3])
 
-        cls.test_dir = tempfile.mkdtemp()
-        cls.path1 = cls.test_dir + "/test_electrodes1.nwb"
-
-        cls.nwbfile1 = NWBFile("session_description1", "file_id1", datetime.now())
+    def setUp(self):
+        """Start with a fresh NWBFile, ElectrodeTable, and remapped BaseRecordings each time."""
+        self.nwbfile = NWBFile(
+            session_description="session_description1", identifier="file_id1", session_start_time=datetime.now()
+        )
+        channel_ids = self.base_recording.get_channel_ids()
+        self.recording_1 = self.base_recording.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=["a", "b", "c", "d"]
+        )
+        self.recording_2 = self.base_recording.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=["c", "d", "e", "f"]
+        )
 
-        cls.RX1 = generate_recording(num_channels=4, durations=[3])
-        cls.metadata_list = [dict(Ecephys={i: dict(name=i, description="desc")}) for i in ["es1", "es2"]]
+    def test_channel_names(self):
+        """Ensure channel names merge correctly after appending."""
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
 
-        channel_ids = cls.RX1.get_channel_ids()
-        num_channels = cls.RX1.get_num_channels()
-        non_int_channel_ids1 = ["a", "b", "c", "d"]
-        non_int_channel_ids2 = ["c", "d", "e", "f"]
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), ["a", "b", "c", "d", "e", "f"])
 
-        cls.RX_non_int_channels1 = cls.RX1.channel_slice(
-            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids1
-        )
-        cls.RX_non_int_channels2 = cls.RX1.channel_slice(
-            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids2
-        )
+    def test_common_property_extension(self):
+        """Add a property for a first recording that is then extended by a second recording."""
+        self.recording_1.set_property(key="property", values=["value_1"] * self.num_channels)
+        self.recording_2.set_property(key="property", values=["value_2"] * self.num_channels)
```

Comment:
My leaning here is to change "property" to "common_property" and "property2" to "added_property" for even more expressiveness. Likewise, I would have the following map for their values:
* "value_1" -> "common_property_value_in_recording_1"
* "value_2" -> "common_property_value_in_recording_2"
* "value_3" -> "value_of_added_property"  

I do think that we are discussing aesthetics at this level, or what do you think? can you cut this one with arguments for one side or the other?

Reply:
I will take care of this now, then.